compiler/tests/fixtures/return_u256_from_called_fn_with_args.fe

```diff
@@ -0,0 +1,10 @@
+contract Foo:
+
+    pub def foo(val1: u256, val2: u256) -> u256:
+        return val1 + val2
+
+    pub def jip() -> u256:
+        return 44
+
+    pub def bar() -> u256:
```

Comment:
I'm curious about the decision to **not** include `self` in the call signature.  Here are the arguments I see for both sides.

Including it is a little confusing because it means that you have to strip it out to compute the function signature.  One example of this is the 4byte.directory which uses regexes to parse solidity source files and extract function signatures.

Excluding it makes the `self` variable kind of *magic*.  Admittedly, it is already a little magic in python to begin with, but I have always liked that it isn't any different from other variables in terms of scope.  It's possible also that the `ContractClass.some_method(instance, *args)` call semantics might be useful for something like making library calls.

```python
class MyContract:
    def foo(self):
        return MyLibrary.some_library_function(self)
```

Reply:
>Excluding it makes the self variable kind of magic. Admittedly, it is already a little magic in python to begin with, but I have always liked that it isn't any different from other variables in terms of scope

Yes, I'm on the same page. I'm actually in favor of requiring `self` for instance methods (so, without `self` means its a class method by definition). I discussed that with @g-r-a-n-t very briefly recently and I think the main argument is just simplicity.

I would be very open (and actually in favor) to make that change.

Reply:
I'm open to this change.

One of the things Christoph and have also discussed is adding explicit mutability similar to rust. I think this could work well with a self parameter.

```python
contract Foo:
  pub def bar(self): # can't modify state (equivalent to Solidity's `view` modifier)
    pass

  pub def baz(mut self): # can modify state
    pass

def foobar(): # can't view or modify state (equivalent to Solidity's `pure`modifier). could also be defined in a contract scope, but is defined in the module scope here and can be shared between contracts in this module and outside of it.
  pass
```

Reply:
This has me thinking about re-entrancy protection...

Reply:
I created an issue for it https://github.com/ethereum/fe/issues/96